app/controllers/inventories_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class InventoriesController < ApplicationController
+  before_action :set_inventory, only: %i[destroy]
+
+  # GET /inventories
+  def index
+    @inventory = Inventory.where(user_id: current_user.id)
+  end
+
+  # GET /inventories/new
+  def new
+    @ingredients_list = ingredients_for_select
+    @inventory = Inventory.new
+  end
+
+  # POST /inventories
+  def create
+    @inventory = Inventory.new(inventory_params.merge(user_id: current_user.id))
+
+    if @inventory.save
+      redirect_to inventories_url, notice: "The ingredient was successfully added to your inventory!"
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /inventories/1 or /inventories/1.json
+  def destroy
+    @inventory.destroy
+    redirect_to inventories_url, notice: "The ingredient was removed from your inventory."
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_inventory
+    @inventory = Inventory.find(params[:id])
+  end
+
+  # Only allow a list of trusted parameters through.
+  def inventory_params
+    params.require(:inventory).permit(:user_id, :ingredient_id)
+  end
+
+  def ingredients_not_in_inventory
+    inventory = Inventory.where(user_id: current_user.id)
+    if inventory.empty?
+      Ingredient.all
+    else
+      Ingredient.where.not(id: inventory.pluck(:ingredient_id))
+    end
+  end
```

Comment:
Following my other suggestion, then this method would change it's line 47. But also take a look how this could be changed using a [guard clause](https://anthonygharvey.com/guard-clauses-vs-nested-conditionals).

```ruby
  def ingredients_not_in_inventory
    inventory = @user.inventory
    return Ingredient.all if inventory.nil?

    Ingredient.where.not(id: inventory.pluck(:ingredient_id))
  end
```

Reply:
Guard clause implemented in the latest commit